### PR TITLE
Add documentation in autodiff modules

### LINKED ISSUE #1217: Add docstring to autodiff's modules in backends
And remove corresponding "ignore-rule" in deep source

### PR DESCRIPTION
**Pull request by ninamiolane**
This closes #1217.

### CODE REVIEW

File: geomstats/_backend/autograd/autodiff.py [hunk 1]
```diff
@@ -16,31 +14,43 @@ def detach(x):
...
-    ----------
-    func : callable
-        Function for which the element-wise grad is computed.
+    Returns
+    -------
+    x : array-like
+        Tensor.
     """
-    return _elementwise_grad(func)
+    return x
```
Comment (nguigs) [inline geomstats/_backend/autograd/autodiff.py:23]: why this change?

File: geomstats/_backend/pytorch/autodiff.py [hunk 2]
```diff
@@ -39,45 +63,76 @@ def backward(ctx, grad_output):
...
+    """
     return lambda x: torch_jac(func, x)
 
 
 def value_and_grad(func, to_numpy=False):
-    """Return a function that returns both value and gradient.
+    """Return a function that returns func's value and gradients' values.
 
-    Suitable for use in scipy.optimize
+    Suitable for use in scipy.optimize.
```
Comment (nguigs) [inline geomstats/_backend/pytorch/autodiff.py:102]: ... with `numpy=True`

File: geomstats/_backend/pytorch/autodiff.py [hunk 3]
```diff
@@ -39,45 +63,76 @@ def backward(ctx, grad_output):
...
 
     Parameters
     ----------
     func : callable
-        Function to compute the gradient. It must be real-valued.
+        Function whose value and gradient values
+        will be computed.
+    to_numpy : bool
+        Unused. Here for API consistency.
```
Comment (nguigs) [inline geomstats/_backend/pytorch/autodiff.py:110]: This is actually used below

Comment (nguigs) [reply geomstats/_backend/pytorch/autodiff.py:110]: I actually added that parameter in the docstring in my ongoing PR #1220

File: geomstats/_backend/autograd/autodiff.py (hunk 1 above)
Comment (ninamiolane) [reply geomstats/_backend/autograd/autodiff.py:23]: we do not use it anywhere

File: geomstats/_backend/pytorch/autodiff.py (hunk 3 above)
Comment (ninamiolane) [reply geomstats/_backend/pytorch/autodiff.py]: Done.
